src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -192,7 +209,50 @@ private DenseMatrix initRhs(LfNetwork lfNetwork, EquationSystem equationSystem,
         return factorsByVarConfig;
     }
 
-    public List<SensitivityValue> analyse(Network network, List<SensitivityFactor> factors, OpenLoadFlowParameters lfParametersExt,
+    /**
+     * Return a mapping between the participating element id through its connected bus, and its participation value in the slack distribution
+     * @param lfNetwork
+     * @param loadFlowParameters
+     * @return
+     */
+    private Map<String, Number> getParticipationFactorByBus(Network network, LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters) {
+
+        Map<String, Number> participationFactorByBusMap = new HashMap<>();
+
+        if (loadFlowParameters.isDistributedSlack()) {
+            ActivePowerDistribution.Step step;
+            List<ParticipatingElement> participatingElements;
+
+            switch (loadFlowParameters.getBalanceType()) {
+                case PROPORTIONAL_TO_GENERATION_P_MAX:
+                    step = new GenerationActionPowerDistributionStep();
+                    break;
+                case PROPORTIONAL_TO_LOAD:
+                    step = new LoadActivePowerDistributionStep(false, false);
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Balance type not yet supported: " + loadFlowParameters.getBalanceType());
+            }
+
+            participatingElements = step.getParticipatingElements(lfNetwork);
+            ParticipatingElement.normalizeParticipationFactors(participatingElements, "bus");
+
+            participatingElements.forEach(participatingElement -> {
+                String busId;
+                if (participatingElement.getElement() instanceof LfGenerator) {
+                    busId = network.getGenerator(((LfGenerator) participatingElement.getElement()).getId()).getTerminal().getBusView().getBus().getId();
```

Comment:
We should avoid searching generator into IIDM model. We could just get LfBus from LfGenerator.

Reply:
Indeed there is no existing method, you mean that we have to implement the getter on all LfGenerator or there is something easier?

Reply:
Yes

Reply:
I have tried something in [the latest commit](https://github.com/powsybl/powsybl-open-loadflow/pull/178/commits/c4342720a2a8c6388f03aac3da1af2cd8f6a329b), tell me if this works for you: